cf/config.go

```diff
@@ -73,12 +74,15 @@ func (c *ClientConfiguration) Validate() error {
 }
 
 // NewConfig creates ClientConfiguration from the provided environment
-func NewConfig(env env.Environment) (*Settings, error) {
+func NewConfig(env env.Environment, settings *sbproxy.Settings) (*Settings, error) {
 	cfSettings := &Settings{Cf: DefaultClientConfiguration(), Reg: &reconcile.Settings{}}
 
 	if err := env.Unmarshal(cfSettings); err != nil {
 		return nil, err
 	}
 
+	cfSettings.Reg.Username = settings.Sm.User
```

Comment:
I'd say remove these Username and password from the settings alltogether to avoid confusion